src/TypeShape.SourceGenerator/Helpers/RoslynHelpers.AdditionalMarkerAttribute.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace TypeShape.SourceGenerator.Helpers;
+
+internal static partial class RoslynHelpers
+{
+    public static void RegisterAdditionalMarkerAttributeName(this IncrementalGeneratorInitializationContext context,
+        MarkerAttributeHolder holder)
+    {
+        var valueProvider = context.AnalyzerConfigOptionsProvider
+            .Select((options, _) =>
+            {
+                if (options.GlobalOptions.TryGetValue(
+                        "build_property.TypeShape_SourceGenerator_AdditionalMarkerAttributeName",
+                        out var markerAttribute))
+                {
+                    NameSyntax attributeNameSyntax = SyntaxFactory.ParseName(markerAttribute);
+                    string attributeName = GetTypeName(attributeNameSyntax, out int attributeArity);
+                    ImmutableArray<string> attributeNamespace = GetNamespaceTokens(attributeNameSyntax);
+                    string? attributeNameMinusSuffix = attributeName.EndsWith("Attribute", StringComparison.Ordinal)
+                        ? attributeName[..^"Attribute".Length]
+                        : null;
+
+                    holder.Populate(attributeName, attributeNameMinusSuffix, attributeNamespace, attributeArity);
+                }
+
+                return 0;
+            });
+        context.RegisterSourceOutput(valueProvider, (_, _) => { }); // this does not produce anything, but is required for the provider to be executed?
```

Comment:
That's surprising, my understanding is that this can only happen if the incremental value somehow doesn't flow to another `RegisterSourceOutput` call.